### PR TITLE
Red Chapter 12 answers

### PR DESCRIPTION
**Pull request by utopia-2**

### CODE REVIEW

File: answers.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+See [TrafficLightDemo](src/org/java/training/TrafficLightDemo.java).
+
+  5) _Define boxing and unboxing. How does autoboxing/unboxing affect these
+      actions?_
+
+Boxing is a mechanism for wrapping Java primitives in Objects to provide OO
+features and allow primitives to be used in Java's limited attempt at generic
+programming. Unboxing is the process of getting a primitive value from a boxed
+Object reference. Autoboxing/unboxing affects these by removing the boilerplate
+associated with these conversions, making them usable.
```
Comment (NormanLittle) [inline answers.md:56]: I'm being _pedantic_ again, but you would have gotten extra credit if you had explicitly said that Autoboxing/unboxing did the conversion automatically (thus removing the associated boilerplate).

File: answers.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	
+	public State selectState() {
+		if ( systemsRunning() ) return ACTIVE;
+		else if ( !diagnosticSuccess() )  return BROKEN;
+		return SUSPENDED;
+	}
+}
+```
+
+Here, we don't have to reference each constant as, e.g. `State.ACTIVE`.
```
Comment (NormanLittle) [inline answers.md:103]: :+1:

File: answers.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+`interface`.
+
+ 11) _What is a marker annotation?_
+
+An annotation without members, for example `@interface BadCode {}`.
+
+ 12) _An annotation can be applied only to methods. True or False?_
+
+False, annotations can be applied to methods, classes, class/instance variables,
+arguments, return types, other annotations - you name it.
```
Comment (NormanLittle) [inline answers.md:116]: Looks good. :+1:

File: src/org/java/training/TrafficLightDemo.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
+package org.java.training;
+
+public class TrafficLightDemo {
+	enum TrafficLight {
+		RED(12_000), GREEN(10_000), AMBER(2_000);
+
+		public final int delay;
+		TrafficLight(int delay) { this.delay = delay; }
+
+		public static TrafficLight next(TrafficLight current) {
```
Comment (NormanLittle) [inline src/org/java/training/TrafficLightDemo.java:10]: Nice! Much neater to have this functionality in here.

File: answers.md (hunk 1 above)
Comment (utopia-2) [inline answers.md:56]: Aww man - can I call that half a point? :star:

**Pull request closed without merging**